### PR TITLE
Fix data race panic 1

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #9897 #9862 

## What this PR does / why we need it:
1. This PR will reduce the occurrence of panic as result of data race.
2. Next PR will completely fix it.

### CODE REVIEW

**Comment by matrix-meow** [comment]
@triump2020 Thanks for your contributions!

The pull request fixes a data race panic issue. The changes made in the pull request include modifying the `newReader` function in `txn_table.go` and adding a `copy` method to the `ModifyBlockMeta` struct in `types.go`. 

There are no issues with the title and body of the pull request. However, there are a few issues with the changes made in the pull request that need addressing:

1. The `copy` method added to the `ModifyBlockMeta` struct is not thread-safe. Since the `ModifyBlockMeta` struct is used in a concurrent environment, the `copy` method should be made thread-safe to avoid data race conditions.

2. The `copy` method creates a new slice and copies the data from the original slice to the new slice. This can be optimized by using the `append` function instead of creating a new slice and copying the data.

3. The `copy` method should be tested to ensure that it works as expected.

4. The pull request does not include any tests to ensure that the data race panic issue is fixed. It is recommended to include tests to ensure that the issue is fixed and to prevent future regressions.

To address these issues, the following changes can be made:

1. To make the `copy` method thread-safe, a mutex can be added to the `ModifyBlockMeta` struct and locked before copying the data. The mutex should be unlocked after the copy is complete.

2. The `copy` method can be optimized by using the `append` function instead of creating a new slice and copying the data. The `append` function can be used to create a new slice and copy the data in a single step.

3. The `copy` method should be tested to ensure that it works as expected. Tests should be added to cover all possible scenarios.

4. Tests should be added to ensure that the data race panic issue is fixed and to prevent future regressions. The tests should cover all possible scenarios and edge cases.

**Comment by triump2020** [comment]
@matrix-meow rerun failed

**Review by nnsgmsone — APPROVED** [review]

**Comment by triump2020** [comment]
@matrix-meow rerun failed

File: pkg/vm/engine/disttae/txn_table.go [hunk 1]
```diff
@@ -1533,12 +1533,17 @@ func (tbl *txnTable) newReader(
 	}
 	for i := 1; i < readerNumber; i++ {
 		if i == readerNumber-1 {
+			//var dst []ModifyBlockMeta
+			dst := make([]ModifyBlockMeta, len(blks[(i-1)*step:]))
```
Comment (fengttt) [inline pkg/vm/engine/disttae/txn_table.go:1537]: This loop is n*n?

Comment (fengttt) [reply pkg/vm/engine/disttae/txn_table.go:1537]: never mind, it is a lop with steps.

Still, what is the cost of ModifyBlockMeta.copy()?   Not sure if it is a problem...

**Comment by triump2020** [comment]
@matrix-meow rerun failed

File: pkg/vm/engine/disttae/txn_table.go (hunk 1 above)
Comment (triump2020) [reply pkg/vm/engine/disttae/txn_table.go:1537]: there' re only a  limited number of blks  whose deletes is empty.

Comment (triump2020) [reply pkg/vm/engine/disttae/txn_table.go:1537]: and next pr will remove these codes.
